### PR TITLE
Only shows a notification when the app is in background.

### PR DESCRIPTION
**Pull request by mootoh**
Obey the behaviour like Google Play Music or Spotify apps.
It is a bit tricky to detect if the whole app goes in background/foreground though...

screenshot

### CODE REVIEW

File: Rebuild/src/main/java/rejasupotaro/rebuild/notifications/PodcastPlayerNotification.java [hunk 1]
```diff
@@ -24,13 +24,17 @@
...
+    }
+
     public static void notify(Context context, Episode episode) {
         notify(context, episode, 0);
     }
 
     public static void notify(Context context, Episode episode, int currentPosition) {
-        if (episode == null || context == null) return;
-
+        if (! isInBackground || episode == null || context == null) return;
```
Comment (rejasupotaro) [inline Rebuild/src/main/java/rejasupotaro/rebuild/notifications/PodcastPlayerNotification.java]: Please remove extra whitespace.

```
if (!isInBackground || episode == null || context == null) return;
```

**Comment by rejasupotaro** [comment]
Could you make a class for observing activity lifecycle.
It is roughly like this.

``` java
package rejasupotaro.rebuild.observers;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import rejasupotaro.rebuild.notifications.PodcastPlayerNotification;

public class ActivityLifecycleObserver implements Application.ActivityLifecycleCallbacks {

    private static Object LOCK = new Object();

    private static ActivityLifecycleObserver INSTANCE = null;

    private int activities = 0;

    private boolean isInBackground = false;

    public static void initialize(Application application) {
        synchronized (LOCK) {
            if (INSTANCE == null) {
                INSTANCE = new ActivityLifecycleObserver();
                application.registerActivityLifecycleCallbacks(INSTANCE);
            }
        }
    }

    public static void terminate(Application application) {
        synchronized (LOCK) {
            if (INSTANCE != null) {
                application.unregisterActivityLifecycleCallbacks(INSTANCE);
                INSTANCE = null;
            }
        }
    }

    @Override
    public void onActivityPaused(Activity activity) {
        activities--;
    }

    @Override
    public void onActivityResumed(Activity activity) {
        if (isInBackground) {
            isInBackground = false;
            PodcastPlayerNotification.setIsInBackground(false);
            PodcastPlayerNotification.cancel(activity);
        }
        activities++;
    }

    @Override
    public void onActivityDestroyed(Activity activity) {
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
    }

    @Override
    public void onActivityStarted(Activity activity) {
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
    }

    @Override
    public void onActivityStopped(Activity activity) {
        if (activities == 0) {
            isInBackground = true;
            PodcastPlayerNotification.setIsInBackground(true);
        }
    }
}
```

And initialize from RebuildApplication.

``` java
public class RebuildApplication extends Application {

    @Override
    public void onCreate() {
        ...
        ActivityLifecycleObserver.initialize(this);
    }

    @Override
    public void onTerminate() {
        ...
        ActivityLifecycleObserver.terminate(this);
        super.onTerminate();
    }
}
```

**Comment by mootoh** [comment]
Copy&Pasted!

**Comment by rejasupotaro** [comment]
LGTM! Thank you for your Pull Request! :+1:

**Pull request merged**